v2/pkg/engine/datasource/grpc_datasource/util.go

```diff
@@ -8,3 +8,42 @@ func initializeSlice[T any](len int, zero T) []T {
 	}
 	return s
 }
+
+// stack is a generic LIFO (Last In First Out) data structure that stores elements of type T.
+type stack[T any] []T
```

Comment:
Is there a better place that we could place this with more "visibility" / "more general" rather than `grpc_datasource/util.go‎`? This is under the assumption we would want other folks to not create another "stack" type again.

Reply:
I changed it because I thought reading large appends and removing indices sometimes and sometimes not was way harder to read than having functions that actually sound like that. I personally think it's way more readable than what I had before